components/Modal/index.d.ts

```diff
@@ -27,14 +24,14 @@ export interface ModalProps {
   };
 }
 
-export default class Modal extends Component<ModalProps> {
-  static Header: ModalHeader;
-
-  static Title: Title;
-
-  static HeaderText: HeaderText;
+interface CompoundedModal extends React.FunctionComponent<ModalProps> {
+  Header: React.FunctionComponent<ModalProps>;
+  Title: React.FunctionComponent<ModalProps>;
+  HeaderText: React.FunctionComponent<ModalProps>;
+  Content: React.FunctionComponent<ModalProps>;
+  Footer: React.FunctionComponent<ModalProps>;
```

Comment:
Have you verified that all types of these subcomponents are functional components? I do not think so

Reply:
Do these subcomponents have the same properties as the parent component? I do not think so

Reply:
If subcomponents are used as class instead of FunctionalComponents the problem will continue